public/app/features/plugins/datasource_srv.ts

```diff
@@ -65,26 +65,17 @@ export class DatasourceSrv implements DataSourceService {
     ref: string | null | undefined | DataSourceRef,
     scopedVars?: ScopedVars
   ): DataSourceInstanceSettings | undefined {
-    const isstring = typeof ref === 'string';
-    let nameOrUid = isstring ? (ref as string) : ((ref as any)?.uid as string | undefined);
-
-    if (nameOrUid === 'default' || nameOrUid === null || nameOrUid === undefined) {
-      if (!isstring && ref) {
-        const type = (ref as any)?.type as string;
-        if (type === ExpressionDatasourceRef.type) {
-          return expressionDatasource.instanceSettings;
-        } else if (type) {
-          console.log('FIND Default instance for datasource type?', ref);
-        }
-      }
-      return this.settingsMapByUid[this.defaultName] ?? this.settingsMapByName[this.defaultName];
-    }
+    let nameOrUid = this.getNameOrUid(ref);
 
     // Expressions has a new UID as __expr__ See: https://github.com/grafana/grafana/pull/62510/
     // But we still have dashboards/panels with old expression UID (-100)
     // To support both UIDs until we migrate them all to new one, this check is necessary
     if (isExpressionReference(nameOrUid)) {
-      return expressionDatasource.instanceSettings;
+      return expressionInstanceSettings;
+    }
+
+    if (nameOrUid === 'default' || nameOrUid === null || nameOrUid === undefined) {
```

Comment:
```suggestion
    if (nameOrUid === 'default' || nameOrUid == null) {
```
personal preference that `==` is the same as checking exact equal on both null and undefined.